Error improvements

rhuanjl opened the pull request:
This PR turns early reference errors into syntax errors per  tc39/ecma262#1527

And makes a few other error message improvements including fixing #6339

fix: #6339

zenparsing commented on lib/Parser/perrors.h:
```diff
@@ -96,7 +96,7 @@ LSC_ERROR_MSG(1084, ERRGetterMustHaveNoParameters, "Getter functions must have n
 LSC_ERROR_MSG(1085, ERRInvalidUseofExponentiationOperator, "Invalid unary operator on the left hand side of exponentiation (**) operator")
 
-LSC_ERROR_MSG(1086, ERRInvalidModuleImportOrExport, "Module import or export statement unexpected here")
+// 1086 -- removed LSC_ERROR_MSG(1086, ERRInvalidModuleImportOrExport, "Module import or export statement unexpected here")
```
We should probably either remove the commented code completely, or reuse this error number (1086) for the  `ERRModuleImportOrExportInScript` error.

rhuanjl replied:
There was a comment at t top of the file about not reusing error numbers and also further up there’s a commented out removed error so I was following what I perceived to be the pattern.

I figure I can reuse it as one of the two I’ve split it into though - will update later.

zenparsing replied:
Ah, you're right!

Let's reuse that one so we can avoid the cruft as much as possible. This is really great by the way!

rhuanjl commented:
I wanted to sort out the ReferenceError/SyntaxError thing to get down to only one test262 fail for me Top Level Await implementation; and thought I may as well do a few more error fixes whilst I was at it. Anyway I've updated for feedback and rebased so hopefully good to merge.
